### PR TITLE
cmd/gist-logs: On 404, the PAT probably needs more permissions

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- This 404 error is from GitHub, and GitHub doesn't give us any more informations about scopes for the gist endpoint (for some reason). But we can safely assume, as it happens a lot, that "Error: Not Found" (404) is because the user hasn't granted their `HOMEBREW_GITHUB_API_TOKEN` the `gist` scope.
- Fixes https://github.com/Homebrew/brew/issues/13988.

Before:

```shell
$ HOMEBREW_GITHUB_API_TOKEN=<token_without_gist_scope> brew gist-logs -p logrotate
Error: Not Found
```

After:

```shell
❯ HOMEBREW_GITHUB_API_TOKEN=<token_without_gist_scope> brew gist-logs logrotate
Error: Your GitHub API token likely doesn't have the `gist` scope.
Create a GitHub personal access token:
https://github.com/settings/tokens/new?scopes=gist&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.zshrc
```

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/utils/github/api.rb [hunk 1]
```diff
@@ -5,7 +5,18 @@
 require "utils/shell"
 require "utils/formatter"
 
+# A module that interfaces with GitHub, code like PAT scopes, credential handling and API errors.
 module GitHub
+  def self.pat_blurb(scopes)
```
Comment (issyl0) [inline Library/Homebrew/utils/github/api.rb:10]:
```suggestion
  def self.pat_blurb(scopes = ALL_SCOPES)
```

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/utils/github/api.rb [hunk 2]
```diff
@@ -49,7 +52,7 @@ def initialize(github_message)
     class RateLimitExceededError < Error
       def initialize(reset, github_message)
         @github_message = github_message
-        new_pat_message = ", or:\n#{CREATE_GITHUB_PAT_MESSAGE}" if API.credentials.blank?
+        new_pat_message = ", or:\n#{pat_blurb}" if API.credentials.blank?
```
Comment (Bo98) [inline Library/Homebrew/utils/github/api.rb:55]:
```suggestion
        new_pat_message = ", or:\n#{GitHub.pat_blurb}" if API.credentials.blank?
```

File: Library/Homebrew/utils/github/api.rb [hunk 3]
```diff
@@ -76,7 +79,7 @@ def initialize(github_message)
             The GitHub credentials in the macOS keychain may be invalid.
             Clear them with:
               printf "protocol=https\\nhost=github.com\\n" | git credential-osxkeychain erase
-            #{CREATE_GITHUB_PAT_MESSAGE}
+            #{pat_blurb}
```
Comment (Bo98) [inline Library/Homebrew/utils/github/api.rb:82]:
```suggestion
            #{GitHub.pat_blurb}
```

File: Library/Homebrew/utils/github/api.rb [hunk 4]
```diff
@@ -87,7 +90,7 @@ def initialize(github_message)
     class MissingAuthenticationError < Error
       def initialize
         message = +"No GitHub credentials found in macOS Keychain or environment.\n"
-        message << CREATE_GITHUB_PAT_MESSAGE
+        message << pat_blurb
```
Comment (Bo98) [inline Library/Homebrew/utils/github/api.rb:93]:
```suggestion
        message << GitHub.pat_blurb
```

File: Library/Homebrew/utils/github/api.rb [hunk 5]
```diff
@@ -177,7 +180,7 @@ def credentials_error_message(response_headers, needed_scopes)
         Your #{what} credentials do not have sufficient scope!
         Scopes required: #{needed_scopes}
         Scopes present:  #{credentials_scopes}
-        #{CREATE_GITHUB_PAT_MESSAGE}
+        #{pat_blurb}
```
Comment (Bo98) [inline Library/Homebrew/utils/github/api.rb:183]:
```suggestion
        #{GitHub.pat_blurb}
```

**Review by Bo98 — COMMENTED** [review]
These are in nested modules/classes so need to be fully qualified.

> Error: undefined local variable or method `pat_blurb' for #<GitHub::API::RateLimitExceededError:0x00007fd0d6af2c98>

**Comment by issyl0** [comment]
Thanks @Bo98, I hadn’t seen the CI failures yet.

**Comment by EricFromCanada** [comment]
Does no one here sleep??

**Comment by Bo98** [comment]
> Does no one here sleep??

That'll be on my todo list somewhere.

**Comment by MikeMcQuaid** [comment]
> > Does no one here sleep??
> 
> That'll be on my todo list somewhere.

@Bo98 I thought we talked about this: add a `help wanted` issue and someone might sleep for you so you don't have to 😂

**Pull request merged**